packages/plugins/RSS3/src/SNSAdaptor/hooks/usePatchFeed.ts

```diff
@@ -0,0 +1,48 @@
+import { Alchemy_EVM, RSS3BaseAPI } from '@masknet/web3-providers'
+import { resolveIPFSLinkFromURL } from '@masknet/web3-shared-evm'
+import { useMemo } from 'react'
+import { useAsyncRetry } from 'react-use'
+import { ChainID } from '../components/FeedCard'
+
+/**
+ *
+ * Patch with some metadata fetched from data provider e.g Alchemy.
+ */
+export function usePatchFeed(feed: RSS3BaseAPI.Web3Feed) {
+    const { value: asset } = useAsyncRetry(async () => {
+        if ((feed.title && feed.summary) || !feed.metadata?.collection_address) return
+
+        const asset = await Alchemy_EVM.getAsset(feed.metadata.collection_address, feed.metadata?.token_id ?? '', {
```

Comment:
Recommend to use hub stead.

Reply:
It's based on old code. I can refactor later in another PR.

Reply:
Please rise a Task issue.